test/e2e/node_maintenance_test.go

```diff
@@ -35,6 +35,8 @@ const (
 	eventInterval         = time.Second * 10
 	timeout               = time.Second * 120
 	testDeployment        = "test-deployment"
+	// sometimes an event is emitted so quic that it is raced with the the current time which is
```

Comment:
I doubt that quick events are the reason. There should be enough time between creating a CR, and the controller reconciling it and emitting the event. I think it's more likely that the time on the controller pod and the test pod are out of sync...?
Also: what about adding the fix to the waitForEvent() function only, instead of everywhere it's called? When we have it at one place only, we also don't need this const with that weird name ;)

Reply:
>I think it's more likely that the time on the controller pod and the test pod are out of sync?

I have checked it locally with the following commands and on AWS cluster bot and there is no significant out-of-sync timing between them.

> ➜   oc exec -it node-maintenance-operator-controller-manager-5bcf79f46-wbpn4 -n openshift-workload-availability -- date -u
Thu Jan 30 15:55:08 UTC 2025
➜ oc exec -it test-deployment-856597466f-j45cs -n node-maintenance-test -- date -u                                      
Thu Jan 30 15:55:12 UTC 2025

Reply:
>Also: what about adding the fix to the waitForEvent() function only, instead of everywhere it's called? When we have it at one place only, we also don't need this const with that weird name ;)

I tried it and it also worked without having this subtle delay..